crates/ruff_db/src/diagnostic/render.rs

```diff
@@ -48,42 +60,71 @@ impl<'a> DisplayDiagnostic<'a> {
         } else {
             AnnotateRenderer::plain()
         };
+        let stylesheet = if config.color {
+            DiagnosticStylesheet::styled()
+        } else {
+            DiagnosticStylesheet::plain()
+        };
         DisplayDiagnostic {
             config,
             resolver,
             annotate_renderer,
             diag,
+            stylesheet,
         }
     }
 }
 
 impl std::fmt::Display for DisplayDiagnostic<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         if matches!(self.config.format, DiagnosticFormat::Concise) {
-            match self.diag.severity() {
-                Severity::Info => f.write_str("info")?,
-                Severity::Warning => f.write_str("warning")?,
-                Severity::Error => f.write_str("error")?,
-                Severity::Fatal => f.write_str("fatal")?,
-            }
+            let (severity, severity_style) = match self.diag.severity() {
+                Severity::Info => ("info", &self.stylesheet.info),
+                Severity::Warning => ("warning", &self.stylesheet.warning),
+                Severity::Error => ("error", &self.stylesheet.error),
+                Severity::Fatal => ("fatal", &self.stylesheet.error),
+            };
+            write_styled(f, severity, severity_style)?;
+
```

Comment:
Could we move `stylesheet into the `fmt` closure (as a local variable). It doesn't seem to be used in any other method and we have access to `config` inside fmt.